components/buildList/js/buildList.controller.js

```diff
@@ -6,6 +6,464 @@ buildApp.controller('buildListController', ['$scope', '$q', '$interval', 'buildL
+			buildPromises.push(buildDefered.promise);
+			updateBuild(build, buildDefered);
+		});
+
+		$q.all(buildPromises).then(function() {
+			var buildStates = [];
+
+			angular.forEach($scope.buildList, function(build) {
+				buildStates.push (build.status);
+			});
+
+			overallStatus = setStatusOfProcess(buildStates);
+			if (overallStatus === 'complete' || overallStatus === 'rejected') {
+				$interval.cancel(updateBuildsInterval);
+			}
+		});
+	}
+
+	init();
+}]).controller('buildListController', ['$scope', '$q', '$interval', 'buildListService', function ($scope, $q, $interval, buildListService) {
+	
+	var deferred = $q.defer(),
+		updateBuildsInterval,
+		intervalTime = 5000,
+		timeElapsed = 0;
+
+	/**
+	 * Initializes the functionality of this controller
+	 *  - requests the data
+	 *  - starts to calculate all the processes
+	 * @method init
+	 */
+	function init() {
+		getData();
+		$q.when(deferred.promise, function() {
+			updateBuildsInterval = $interval(updateBuildsList, intervalTime);
+		});
+	}
+
+	/**
+	 * Requests for the build list data
+	 * @method getData
+	 */
+	function getData() {
+		return buildListService.getData().then(function(response) { 
+			$scope.buildList = response.data;
+			deferred.resolve();
+		});
+	}
+
+	/**
+	 * This is called when details about a specific build or firewall are requested
+	 *  - if the build status is not complete or rejected then we can't display the results
+	 * @method displayResults
+	 * @param  {Object} item Item for which details are requested
+	 * @return {Boolean} if status of item is not complete or rejected, false value is returned otherwise true
+	 */
+	$scope.displayResults = function(item) {
+		if (item.status !== 'complete' && item.status !== 'rejected') {
+			return false;
+		}
+
+		buildListService.getResults(item.id).then(function(response) {
+			item.results = response.data;
+		});
+
+		return true;
+	};
+
+	/**
+	 * Establishes the status of a specific job (metrics, build, unit test, functional test)
+	 * - if the completness percentage it's greater or equal with 100% then a request to the server 
+	 * 	 should have been made in order to establish the final status (I choosed randomly between complete and rejected)
+	 * - if the completnees percentage it's smaller than 100, 'running' status is returned
+	 * @method setStatusOnJob
+	 * @param {Object} job object containing information about the job (percentage, status)
+	 * @param {Number} percentage
+	 */
+	function setStatusOnJob(job, percentage) {
+		if (percentage >= 100) {
+			var random = Math.round(Math.random());
+			job.status = (random) ? 'complete' : 'rejected';
+			job.percentage = 100;
+			return;
+		}
+
+		job.status = 'running';
+		job.percentage = percentage; 
+	}
+
+	/**
+	 * Sets the status of a process after analysing the states of its children
+	 * @method setStatusOfProcess
+	 * @param {Array} statesArr states array of the children of the process
+	 */
+	function setStatusOfProcess(statesArr) { 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3ATrailingWhitespace)